examples/with-next/pages/index.tsx

```diff
@@ -187,79 +191,91 @@ export default class Index extends PureComponent {
     })
   }
 
-  startKeyword = async () => {
-    this.initialize()
-    this.setState({ keyword: { status: 'Calibrating...' } })
-    const profile = PipelineProfile.Keyword
-    profile.speechConfig = {
-      ...profile.speechConfig,
-      baseKeywordUrl: 'https://s.spokestack.io/u/UbMeX/js',
-      keywordClasses: [
-        'zero',
-        'one',
-        'two',
-        'three',
-        'four',
-        'five',
-        'six',
-        'seven',
-        'eight',
-        'nine'
-      ]
-    }
-    profile.onReady = () => {
-      this.setState({ keyword: { status: 'Listening', result: '' } })
-    }
-    profile.onEvent = (evt) => {
-      const { eventType, transcript } = evt
-      if (eventType == EventType.Recognize) {
-        this.setState({ keyword: { error: '', result: transcript } })
-      } else if (eventType == EventType.Timeout) {
-        this.setState({ keyword: { error: 'timeout' } })
+  toggleKeyword = async () => {
+    const isActive = this.state.activeDemo != undefined
+    if (isActive) {
+      this.stopRecording()
+    } else {
+      this.initialize()
+      this.setState({ activeDemo: 'keyword', keyword: { status: 'Calibrating...' } })
+      const speechConfig = {
+        ...DefaultSpeechConfig,
+        keywordClasses: [
+          'zero',
+          'one',
+          'two',
+          'three',
+          'four',
+          'five',
+          'six',
+          'seven',
+          'eight',
+          'nine'
+        ]
       }
-    }
 
-    startPipeline(profile)
+      startPipeline({
+        profile: PipelineProfiles.Keyword,
+        baseUrls: { keyword: 'https://s.spokestack.io/u/UbMeX/js' },
+        speechConfig: speechConfig,
+        onReady: () => {
+          this.setState({ keyword: { status: 'Listening', result: '' } })
+        },
```

Comment:
Remove onReady, make `startPipeline` a promise that resolves when ready, or rejects if it doesn't start.